csrc/multidevice/utils.h

```diff
@@ -86,7 +86,30 @@ bool haveDifferentShardings(
 // Returns whether a resharding expr reshards an inner axis
 bool isInnerResharding(Expr* expr);
 
-void shardAllLike(TensorView* ref, std::vector<TensorView*> tvs);
+// Shards all tensors in tvs like reference
+void shardAllLike(
+    TensorView* ref,
+    std::vector<TensorView*> tvs,
+    const std::unordered_set<ParallelType>& parallel_types = {
+        ParallelType::Serial,
```

Comment:
Why Serial? IIRC, everything starts with Serial and parallelAllLike will make some of them DIDx.

Reply:
I recall we added serial for cases in the reorder sharding axis pass. It might be good to revisit that and see if we should be using shardAllLike in that case.